packages/core/src/extensions/Blocks/api/blocks/createSpec.ts

```diff
@@ -50,8 +55,60 @@ export type CustomBlockImplementation<
     dom: HTMLElement;
     contentDOM?: HTMLElement;
   };
+
+  parse?: (el: HTMLElement) => PartialBlockFromConfig<T, I, S> | undefined;
 };
 
+// Function that uses the 'parse' function of a blockConfig to create a
+// TipTap node's `parseHTML` property. This is only used for parsing content
+// from the clipboard.
+export function getParseRules(
+  config: BlockConfig,
+  customParseFunction: CustomBlockImplementation<any, any, any>["parse"]
+) {
+  const rules: ParseRule[] = [
+    {
+      tag: "div[data-content-type=" + config.type + "]",
+    },
+  ];
+
+  if (customParseFunction) {
+    rules.push({
+      tag: "*",
+      getAttrs(node: string | HTMLElement) {
+        if (typeof node === "string") {
+          return false;
+        }
+
+        const block = customParseFunction?.(node);
+
+        if (block === undefined) {
+          return false;
+        }
+
+        return block.props || {};
+      },
+    });
+  }
+  //     getContent(node, schema) {
```

Comment:
I'm kind of wondering if this use case and API makes sense. I'm not really sure if I can see any scenarios where someone would want to override the default inline content parsing, especially since if you can provide a parse function for custom inline content.

Reply:
Ok, let's not add it if we don't have a use-case yet